ogcapi-stable/ogcapi-features-html/src/main/java/de/ii/ldproxy/ogcapi/features/html/app/FeatureEncoderHtml.java

```diff
@@ -0,0 +1,238 @@
+          if (i == transformationContext.getPage()) {
+            pagination.add(new NavigationDTO(String.valueOf(i), true));
+          } else {
+            pagination.add(new NavigationDTO(String.valueOf(i), String.format("limit=%d&offset=%d", transformationContext.getLimit(), (i - 1) * transformationContext.getLimit())));
+          }
+        }
+
+        if (transformationContext.getPage() < pages) {
+          pagination
+              .add(new NavigationDTO("›", String.format("limit=%d&offset=%d", transformationContext.getLimit(), transformationContext.getOffset() + transformationContext.getLimit())))
+              .add(new NavigationDTO("»", String.format("limit=%d&offset=%d", transformationContext.getLimit(), (pages - 1) * transformationContext.getLimit())));
+          metaPagination
+              .add(new NavigationDTO("next", String.format("limit=%d&offset=%d", transformationContext.getLimit(), transformationContext.getOffset() + transformationContext.getLimit())));
+        } else {
+          pagination
+              .add(new NavigationDTO("›"))
+              .add(new NavigationDTO("»"));
+        }
+      } else {
+        int from = Math.max(1, transformationContext.getPage() - 2);
+        int to = transformationContext.getPage();
+        for (int i = from; i <= to; i++) {
+          if (i == transformationContext.getPage()) {
+            pagination.add(new NavigationDTO(String.valueOf(i), true));
+          } else {
+            pagination.add(new NavigationDTO(String.valueOf(i), String.format("limit=%d&offset=%d", transformationContext.getLimit(), (i - 1) * transformationContext.getLimit())));
+          }
+        }
+        if (returned >= transformationContext.getLimit()) {
+          pagination
+              .add(new NavigationDTO("›", String.format("limit=%d&offset=%d", transformationContext.getLimit(), transformationContext.getOffset() + transformationContext.getLimit())));
+          metaPagination
+              .add(new NavigationDTO("next", String.format("limit=%d&offset=%d", transformationContext.getLimit(), transformationContext.getOffset() + transformationContext.getLimit())));
+        } else {
+          pagination
+              .add(new NavigationDTO("›"));
+        }
+      }
+
+      transformationContext.collection().pagination = pagination.build();
+      transformationContext.collection().metaPagination = metaPagination.build();
+
+    } else if (transformationContext.isFeatureCollection()) {
+      LOGGER.error("Pagination not supported by feature provider, the number of matched items was not provided.");
+    }
+  }
+
+  @Override
+  public void onFeature(FeatureHtml feature) {
+    transformationContext.featuresHtmlConfiguration().getFeatureTitleTemplate().ifPresent(label -> transformationContext.collection().name = label);
+
+    //TODO
+    if (!feature.hasObjects()) {
+      feature.getProperties().forEach(propertyHtml -> propertyHtml.name(propertyHtml.getSchema().map(
```

Comment:
I think this should be

```
feature.getProperties().forEach(propertyHtml -> propertyHtml.name(propertyHtml.getSchema().flatMap(
          FeatureSchema::getLabel).orElseGet(propertyHtml::getName)).propertyPath(ImmutableList.of(propertyHtml.getName())));
```

Reply:
This should have catched only the classic/flatten case, which does not use labels. But it was also applied to schemas that were flat in the first case. I implemented a better solution.